packages/typespec-ts/src/modular/buildCodeModel.ts

```diff
@@ -1591,7 +1591,10 @@ function emitType(
       return emitEnum(context, type);
     case "EnumMember":
       return emitEnumMember(context, type);
+    case "ModelProperty":
```

Comment:
Could you add a UT for case like https://github.com/v-jiaodi/autorest.typescript/blob/update-todo-smoke-test/packages/typespec-test/test/todo_non_branded/spec/main.tsp#L225 in models/deserialization/propertyType.md?